crates/ra_analysis/src/imp.rs

```diff
@@ -220,6 +220,34 @@ impl AnalysisImpl {
         let source_root = self.db.file_source_root(file_id);
         self.db.module_tree(source_root)
     }
+    pub(crate) fn module_path(&self, position: FilePosition) -> Cancelable<String> {
+        let mut current_position = position;
+        let mut path = "".to_string();
+
+        loop {
+            println!("current_position {:?}", current_position);
+            if let Some((file_id, file_symbol)) = self.parent_module(current_position)?.first() {
```

Comment:
Hm, calling `Analysis::parent_module` is probably not the right thing to do here: this function is an API function used by LSP to implement "go to super module" feature. 

We should add a `path_to_root` method to `ModuleId` instead, here:

https://github.com/rust-analyzer/rust-analyzer/blob/923483e321acace3bbf38688bd70d4d38f49b35e/crates/ra_analysis/src/descriptors/module/mod.rs#L82

then, we should use this method directly here, but constructing ModuleSource, converting it to ModuleId, etc, like we do in the implementation of Analysis::parent_module

Reply:
> this function is an API function used by LSP to implement "go to super module" feature.

This explains why, for inline modules, it returns the location of module's own identifier: this mirrors the way it woks for non-inline modules.

Reply:
I try to get a the module name by the ModuleId and it seems a bit problematic.

Reply:
Take a look at #234: it introduces a more OO-flavored API for modules, which hopefully should be easier to use.

In particular, a `ModuleDescriptor` now has  a `::name` method.